core/src/executor/sort.rs

```diff
@@ -41,52 +47,111 @@ impl<'a> Sort<'a> {
 
     pub async fn apply(
         &self,
-        rows: impl Stream<Item = Result<AggregateContext<'a>>> + 'a,
-    ) -> Result<Pin<Box<dyn Stream<Item = Item<'a>> + 'a>>> {
-        if self.order_by.is_empty() {
-            let rows = rows.map_ok(|aggregate_context| {
-                let AggregateContext { aggregated, next } = aggregate_context;
+        rows: impl Stream<Item = Result<(AggregateContext<'a>, Row)>> + 'a,
+        labels: Rc<Vec<String>>,
+        table_alias: &'a str,
+    ) -> Result<impl Stream<Item = Result<Row>> + 'a> {
+        #[derive(futures_enum::Stream)]
+        enum Rows<I1, I2> {
+            NonOrderBy(I1),
+            OrderBy(I2),
+        }
 
-                (aggregated.map(Rc::new), next)
-            });
+        if self.order_by.is_empty() {
+            let rows = rows.map_ok(|(_, row)| row);
 
-            return Ok(Box::pin(rows));
+            return Ok(Rows::NonOrderBy(Box::pin(rows)));
         }
-
         let rows = rows
-            .and_then(move |AggregateContext { aggregated, next }| async move {
-                let blend_context = Rc::clone(&next);
+            .and_then(|(AggregateContext { aggregated, next }, row)| {
+                enum SortType<'a> {
+                    Value(Value),
+                    Expr(&'a Expr),
+                }
+
+                let order_by = self
+                    .order_by
```

Comment:
Could we.. have some more line breaks here? It would be better to read if we have some more line breaks.

```rust
        if self.order_by.is_empty() {
            let rows = rows.map_ok(|(_, row)| row);

            return Ok(Rows::NonOrderBy(Box::pin(rows)));
        }

        let rows = rows
            .and_then(|(AggregateContext { aggregated, next }, row)| {
                enum SortType<'a> {
                    Value(Value),
                    Expr(&'a Expr),
                }

                let order_by = self
                    .order_by
                    .iter()
                    .map(|OrderByExpr { expr, asc }| -> Result<_> {
                        let big_decimal = match expr {
                            Expr::Literal(AstLiteral::Number(n)) => Some(n),
                            Expr::UnaryOp {
                                op: UnaryOperator::Plus,
                                expr,
                            } => match expr.as_ref() {
                                Expr::Literal(AstLiteral::Number(n)) => Some(n),
                                _ => None,
                            },
                            _ => None,
                        };

                        match big_decimal {
                            Some(n) => {
                                let index = n
                                    .to_usize()
                                    .ok_or_else(|| -> Error { SortError::Unreachable.into() })?;
                                let zero_based = index.checked_sub(1).ok_or_else(|| -> Error {
                                    SortError::ColumnIndexOutOfRange(index).into()
                                })?;
                                let value = row.get_value(zero_based).ok_or_else(|| -> Error {
                                    SortError::ColumnIndexOutOfRange(index).into()
                                })?;

                                Ok((SortType::Value(value.clone()), *asc))
                            }
                            None => Ok((SortType::Expr(expr), *asc)),
                        }
                    })
                    .collect::<Result<Vec<_>>>();

                let labels = Rc::from(labels.as_slice());
                let filter_context = Rc::new(FilterContext::concat(
                    self.context.as_ref().map(Rc::clone),
                    Some(Rc::clone(&next)),
                ));
                let aggregated = aggregated.map(Rc::new);

                async move {
                    let row = Rc::new(row);
                    let label_context = Rc::from(BlendContext::new(
                        table_alias,
                        labels,
                        Shared(Rc::clone(&row)),
                        None,
                    ));
                    let filter_context = Rc::new(FilterContext::concat(
                        Some(filter_context),
                        Some(Rc::clone(&label_context)),
                    ));

                    let values = stream::iter(order_by?.into_iter())
                        .then(|(sort_type, asc)| {
                            let context = Some(Rc::clone(&filter_context));
                            let aggregated = aggregated.as_ref().map(Rc::clone);

                            async move {
                                let value = match sort_type {
                                    SortType::Value(value) => value,
                                    SortType::Expr(expr) => {
                                        evaluate(self.storage, context, aggregated, expr)
                                            .await?
                                            .try_into()?
                                    }
                                };

                                Ok::<Value, Error>((value, asc))
                            }
                        })
                        .try_collect::<Vec<_>>()
                        .await?;

                    drop(label_context);
                    drop(filter_context);

                    let row = Rc::try_unwrap(row).map_err(|_| SortError::Unreachable)?;

                    Ok((values, row))
                }
            })
```